rpcserver.go

```diff
@@ -1059,9 +1059,65 @@ func (r *rpcServer) ListAssets(ctx context.Context,
 			"and include_leased")
 	}
 
+	constraints := tapfreighter.CommitmentConstraints{
+		MinAmt:         req.MinAmount,
+		MaxAmt:         req.MaxAmount,
+		CoinSelectType: tapsend.ScriptTreesAllowed,
+	}
+
+	if req.GroupKey != nil {
+		groupKey, err := btcec.ParsePubKey(req.GroupKey[:])
+		if err != nil {
+			return nil, fmt.Errorf("error parsing group key: %w",
+				err)
+		}
+
+		// Construct an asset specifier using the requested group key.
+		groupSpecifier, err := asset.NewSpecifier(
+			nil, groupKey, nil, false,
+		)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"error creating asset specifier: %w",
+				err)
+		}
+
+		constraints.AssetSpecifier = groupSpecifier
+	}
+
+	filters := &tapdb.AssetQueryFilters{
+		CommitmentConstraints: constraints,
+	}
+
+	if req.ScriptKey != nil {
```

Comment:
Same here, should be length check, not `nil` check.

Reply:
This results in a compiler error:

```
./rpcserver.go:1082:9: invalid argument: req.ScriptKey (variable of type *taprpc.ScriptKey) for built-in len
```

Other RPC methods such as `ExportProofRequest` define `ScriptKey` as bytes whereas `ListAssetRequest` uses the tap RPC `ScriptKey` struct.

Should `ListAssetRequest` be updated to use bytes instead of the `ScriptKey` struct? Or is there a different way I should approach this?

Reply:
No, it's fine. Didn't see this was a struct when reviewing. So the `nil` check is okay, the `taprpc.UnmarshalScriptKey` will do the required length checks.